src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectionTestWithSSLCert/CertificateTestWithTdsServer.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Data;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using Microsoft.Data.SqlClient.ManualTesting.Tests.DataCommon;
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.ManualTesting.Tests
+{
+    public class CertificateTestWithTdsServer
+    {
+        private static readonly string s_fullPathToPowershellScript = Path.Combine(Directory.GetCurrentDirectory(), "makepfxcert.ps1");
+        private static readonly string s_fullPathToPfx = Path.Combine(Directory.GetCurrentDirectory(), "localhostcert.pfx");
+        private static readonly string s_fullPathToClientCert = Path.Combine(Directory.GetCurrentDirectory(), "clientcert");
+
+        public CertificateTestWithTdsServer()
+        {
+            if (!Directory.Exists(s_fullPathToClientCert))
+            {
+                Directory.CreateDirectory(s_fullPathToClientCert);
+            }
```

Comment:
Fail if user does not have elevated access on Windows
you can add something similar to below:
```C#
 // Confirm that user has elevated access on Windows
 if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
 {
     using WindowsIdentity identity = WindowsIdentity.GetCurrent();
     WindowsPrincipal principal = new(identity);
     if (!principal.IsInRole(WindowsBuiltInRole.Administrator))
     {
         Assert.Fail("User needs to have elevated access for these set of tests");
     }
 }
```

Also, is the created certificate disposed at the end?

Reply:
Added Assertion if not running as Administrator as suggested.

Cleaning up of certificates were implemented as suggested.